recipe/mpiexec.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -euo pipefail
+# pipe stdout, stderr through cat to avoid O_NONBLOCK issues
+exec mpiexec "$@" 2>&1</dev/null | cat
```

Comment:
In the future, I think we can remove this. It's supposedly been fixed in mpich 3.2.1, but no harm keeping it for now.

Reply:
Yes.  After this PR is merged, we have to update to mpich 3.3.1, I was planning to do it by then.

Or do you prefer to jump now to 3.3.1 and forget about getting fresh builds for 3.2.1?

Reply:
In the openmpi recipe, we still need the script to pass the `--allow-run-as-root`. Maybe it is a good idea to keep the mpiexex script in mpich as well, as a future last resort in case we are in need of similar tweaks.

Reply:
Sure. No real difference between the script and the function that was here before. Still some arguments to consolidate. Without redirecting, an alias in `run_test.sh` ought to be enough.

Reply:
However, the use the function that was here before was broken, better to use a script in which we can control bash options. See this log https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=45392 and look for `./helloworld_c: error while loading shared libraries: libgfortran.so.4: cannot open shared object file: No such file or directory`. So eventually the PR was merged but the package was broken.